src/docs/sphinx/tutorials/step12/Tutorial.rst

```diff
@@ -0,0 +1,357 @@
+
+To specify a coupling between two different solvers, we define and characterize each single-physics solver separately.
+Then, we customize a *coupling solver* between these single-physics
+solvers as an additional solver.
+This approach allows for generality and flexibility in constructing multi-physics solvers.
+The order of specifying these solvers is not restricted in GEOSX.
+Note that end-users should give each single-physic solver a meaningful and distinct name, as GEOSX will recognize these single-physics solvers based on their customized names and create user-expected coupling.
+
+As demonstrated in this tutorial, to setup a poro-mechanical coupling, we need to effectively define three different solvers in the XML file:
+
+- the small strain Lagrangian mechanics solver, a solver of type ``SolidMechanics_LagrangianFEM`` called here ``mechanicsSolver`` (more information here: :ref:`SolidMechanicsLagrangianFEM`),
+
+.. literalinclude:: ../../../../../examples/plasticity/WellboreProblem_PoroDruckerPrager.xml
+  :language: xml
+  :start-after: <!-- SPHINX_WELLBORE_MECHANICALSOLVER -->
+  :end-before: <!-- SPHINX_WELLBORE_MECHANICALSOLVER_END -->
+
+
+- the single-phase flow solver, a solver of type ``SinglePhaseFVM`` called here ``SinglePhaseFlowSolver`` (more information on these solvers at :ref:`SinglePhaseFlow`),
+
+.. literalinclude:: ../../../../../examples/plasticity/WellboreProblem_PoroDruckerPrager.xml
+  :language: xml
+  :start-after: <!-- SPHINX_WELLBORE_SINGLEPHASEFVM -->
+  :end-before: <!-- SPHINX_WELLBORE_SINGLEPHASEFVM_END -->
+
+
+- the coupling solver (``Poroelastic``) that will bind the two single-physics solvers above, which is named as ``PoroelasticitySolver`` (more information at :ref:`PoroelasticSolver`).
+
+.. literalinclude:: ../../../../../examples/plasticity/WellboreProblem_PoroDruckerPrager.xml
+  :language: xml
+  :start-after: <!-- SPHINX_WELLBORE_POROELASTICSOLVER -->
+  :end-before: <!-- SPHINX_WELLBORE_POROELASTICSOLVER_END -->
+
+
+The two single-physics solvers are parameterized as explained
+in their corresponding documents. 
+
+In this tutorial, let us focus on the coupling solver.
+This solver (``PoroelasticitySolver``) uses a set of attributes that specifically describe the coupling process within a poromechanical framework.
+For instance, we must point this solver to the designated fluid solver (here: ``SinglePhaseFlowSolver``) and solid solver (here: ``mechanicsSolver``).
+These solvers are forced to interact through the ``porousMaterialNames="{porousRock}"`` with all the constitutive models. We specify the the discretization method (``FE1``, defined in the ``NumericalMethods`` section), and the target regions (here, we only have one, ``Omega``).
+And more parameters are required to characterize a coupling procedure (more information at :ref:`PoroelasticSolver`). In this way, the two single-physics solvers will be simultaneously called and executed for solving the wellbore problem here.
```

Comment:
English style point: sentences should never begin with `And`.  You can simply say, "More parameters ..." or "In addition, extra parameters ..."

Reply:
Agree